server/controller/src/main/java/ai/starwhale/mlops/domain/job/status/JobUpdateHelper.java

```diff
@@ -90,6 +93,8 @@ public void updateJob(Job job) {
                     TaskStatusChangeWatcher.SKIPPED_WATCHERS.remove();
                 });
             }
+            // deal with the data which belong to current job TODO: the same with retry logic
+            dataLoader.resetUnProcessed(job.getUuid());
```

Comment:
Is it better to reset unprocessed data at task level. 
Currentlly job fails if at least one task fails, this logic is not proper in consumption use cases indeed.  The proper logic shall be success at least one success.
Reseting unprocessed data at task level could make us easily correct the status propergation logic.
Moreover, job doesn't necessarily have a dataset consumption.
So better to move this logic to task watcher?

Reply:
done